src/ol/interaction/modifyinteraction.js

```diff
@@ -85,6 +85,14 @@ ol.interaction.Modify = function(options) {
   this.lastPixel_ = [0, 0];
 
   /**
+   * Keep track of the last inserted pixel location to avoid
+   * unintentional deletion.
+   * @type {ol.Pixel}
+   * @private
+   */
+  this.lastNewVertexPixel_ = [-1, -1];
```

Comment:
I only noticed the checks below after I send my comment. You are right, `[NaN, NaN]` is cleaner because you don't have to do an additional check (`goog.isNull`).